docs/training/apis-howto/api-core-mnist-ug.rst

```diff
@@ -0,0 +1,573 @@
+   </training/hyperparameter/search-methods/overview>`.
+
+-  ``metric``: ``test_loss``
+
+-  ``smaller_is_better``: ``True`` (This is equivalent to minimization vs. maximization of
+   objective.)
+
+-  ``max_trials``: 500 (This is the maximum number of trials the searcher should run.)
+
+-  ``max_length``: 20 epochs (The max length of a trial. For more information, visit Adaptive ASHA
+   in the :doc:`Experiment Configuration Reference
+   </reference/reference-training/experiment-config-reference>`.
+
+In addition, we also need to define the hyperparameters themselves. Adaptive ASHA will pick values
+between the ``minval`` and ``maxval`` for each hyperparameter for each trial.
+
+..
+   note:
+
+   To see early stopping in action, try setting ``max_trials`` to over 500 and playing around with the hyperparamter search values.
+
+In this step, we’ll run our experiment using the model_def_adaptive.py script and its accompanying
+adaptive.yaml experiment configuration file.
+
+Begin by accessing the hyperparameters in your code:
+
+..
+   note:
+
+   Refer to line 204 in ``model_def_adaptive.py``
+
+.. code::
+
+   #NEW: Get hparams chosen for this trial from cluster info object.
+   hparams = info.trial.hparams
+
+.. image:: /assets/images/model-def-adaptive-script.png
+
+Then, pass the hyperparameters into your model and optimizer:
+
+.. code::
+
+   #NEW: Pass relevant hparams to model and optimizer.
+   model = Net(hparams).to(device) optimizer =
+   optim.Adadelta(model.parameters(), lr=hparams[“learning_rate”])
+
+Ensure your model is set to use the selected values on a per-trial basis rather than your previously
+hardcoded values:
+
+..
+   note:
+
+   Refer to line 18 in ``model_def_adaptive.py``
+
+.. code::
+
+   class Net(nn.Module):
+       # NEW: Add hparams to **init**.
+       def **init**\ (self, hparams):
+           # NEW: Read hyperparameters provided for this trial.
+           super(Net, self).__init()
+           self.conv1 = nn.Conv2d(1,hparams[“n_filters1”], 3, 1)
+           self.conv2 = nn.Conv2d(hparams[“n_filters1”], hparams[“n_filters2”], 3, 1)
+           self.dropout1 = nn.Dropout(hparams[“dropout1”])
+           self.dropout2 = nn.Dropout(hparams[“dropout2”])
+           self.fc1 = nn.Linear(144 * hparams[“n_filters2”], 128)
+           self.fc2 = nn.Linear(128, 10)
+
+Step 4.1: Run the Experiment
+============================
+
+Run the following command to run the experiment:
+
+.. code::
+
+   det e create adaptive.yaml .
+
+In the Determined WebUI, navigate to the **Hyperparameters** tab.
+
+You should see a graph in the WebUI that displays the various trials initiated by the Adaptive ASHA
+hyperparameter search algorithm.
+
+******************************
+ Step 5: Distributed Training
+******************************
+
+The Core API has special considerations for running distributed training. Some of the more important
+considerations are:
```

Comment:
I think this was my own writing, but it's not phrased well.

"Special considerations" does not communicate whether we're talking about a feature that Core API _gives to the user_ or a requirement that Core API _places on the user_.

Recommend rewording "considerations" -> "features".

Reply:
makes sense